### PR TITLE
Added inspection for mixed named and unnamed parameters (fixes #2494)

### LINKED ISSUE #2494: New inspection: "Return value name required"
This code doesn't compile:
```go
func mixNamedAndUnnamedReturnValues() (a string, error) {
	return
}
```

I'd like the plugin to highlight this code red. 

I like it that ```error``` is parsed as type here (the compiler parses it as variable name) - it allows navigating to types while working on implementation. So I propose adding an inspection which would complain on unnamed parameters.

Note that this code is valid, though:
```go
func mixNamedAndUnnamedReturnValues() (string, a json.Decoder) {
	return
}
```

Plugin version: 0.11.1316

### PR DESCRIPTION
**Pull request by dlsniper**
The inspection might need a better reporting target. As for a quick fix, I'm not sure how to proceed on this as there are different issues with each approach (make everything named, make everything unnamed).

### CODE REVIEW

File: src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      @Override
+      public void visitMethodDeclaration(@NotNull GoMethodDeclaration o) {
+        super.visitMethodDeclaration(o);
+        visitDeclaration(holder, o.getSignature(), "Method");
+      }
+
+      @Override
+      public void visitFunctionDeclaration(@NotNull GoFunctionDeclaration o) {
+        super.visitFunctionDeclaration(o);
+        if (GoTestFinder.isTestFile(o.getContainingFile()) && GoTestFunctionType.fromName(o.getName()) != null) {
```
Comment (zolotov) [inline src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java]: Why?

Comment (zolotov) [inline src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java]: Test for this?

File: src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    for (GoParameterDeclaration parameterDeclaration : parameters.getParameterDeclarationList()) {
+      if (parameterDeclaration.getParamDefinitionList().isEmpty()) {
+        hasUnnamed = true;
+      }
+      else {
+        hasNamed = true;
+      }
+    }
+
+    if (hasNamed && hasUnnamed) {
```
Comment (zolotov) [inline src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java]: Might be inside `for`

File: src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java (hunk 1 above)
Comment (dlsniper) [inline src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java]: I've done a stupid copy paste and forgot to think for that, will remove, thanks.

File: src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java (hunk 2 above)
Comment (dlsniper) [inline src/com/goide/inspections/GoMixedNamedUnnamedParametersInspection.java]: What I've noticed while working on this was that there's a difference between the Go parser and the plugin parser for the following Parameters ` a, b int, string, int `:
- the plugin parser sees it as: ` a, b int // ParamDecl1, string // unamed ParamDecl2, int // unamed ParamDecl3 ` 
- the Go parser sees it as: ` a, b int // ParamDecl1, string, int // ParamDecl2 + missing type info `

I can move it in the for loop but it also depends on how the reporting should be done for this. Should all the parameter/return list be flagged?

The quick fix for this one is not immediately clear, maybe the quickfix should be postponed until the change signature capability lands (if ever done at all)?

**Comment by zolotov** [comment]
merged manually

**Pull request closed without merging**